source/slang/slang-ir-insts.h

```diff
@@ -296,25 +296,23 @@ struct IREntryPointDecoration : IRDecoration
     Profile getProfile() { return Profile(Profile::RawVal(GetIntVal(getProfileInst()))); }
 };
 
-struct IRGeometryPrimitiveTypeDecoration: IRDecoration
+struct IRGeometryInputPrimitiveTypeDecoration: IRDecoration
 {
-    enum { kOp = kIROp_GeometryPrimitiveTypeDecoration };
-    IR_PARENT_ISA(GeometryPrimitiveTypeDecoration)
+    IR_PARENT_ISA(GeometryInputPrimitiveTypeDecoration)
 };
 
-IR_SIMPLE_DECORATION(PointPrimitiveTypeDecoration)
-IR_SIMPLE_DECORATION(LinePrimitiveTypeDecoration)
-IR_SIMPLE_DECORATION(TrianglePrimitiveTypeDecoration)
-IR_SIMPLE_DECORATION(LineAdjPrimitiveTypeDecoration)
-IR_SIMPLE_DECORATION(TriangleAdjPrimitiveTypeDecoration)
+IR_SIMPLE_DECORATION(PointInputPrimitiveTypeDecoration)
+IR_SIMPLE_DECORATION(LineInputPrimitiveTypeDecoration)
+IR_SIMPLE_DECORATION(TriangleInputPrimitiveTypeDecoration)
+IR_SIMPLE_DECORATION(LineAdjInputPrimitiveTypeDecoration)
+IR_SIMPLE_DECORATION(TriangleAdjInputPrimitiveTypeDecoration)
 
     /// This is a bit of a hack. The problem is that when GLSL legalization takes place
-    /// A decoration that marks a value as having linkage.
-    ///
-    /// A value with linkage is either exported from its module,
-    /// or will have a definition imported from another module.
-    /// In either case, it requires a mangled name to use when
-    /// matching imports and exports.
+    /// the parameters from the entry point are globalized *and* potentially split
+    /// So even if we did copy a suitable decoration onto the globalized parameters,
+    /// it would potentially output multiple times without extra logic.
+    /// Using this decoration we can copy the StreamOut type to the entry point, and then
+    /// emit as part of entry point attribute emitting.  
 struct IRStreamOutputTypeDecoration : IRDecoration
```

Comment:
Not relevant to this PR, but eventually we might want to factor out the input/output primitive topologies to just be `enum`s, and then have these decorations either take the `enum` as an operand, or be encoded so that their opcode is `baseOpcode + enumVal`. That would potentially simplify things so that we don't have things like the indirection here, where we have a decoration on an entry point that references an IR type that might not even be used/required by the output.